cdragontoolbox/binfile.py

```diff
@@ -361,6 +402,33 @@ def read_binfile_entry(self, htype):
         assert self.f.tell() - pos == length
         return entry
 
+    def read_patch_section(self):
+        """Reads the entire PTCH section"""
+
+        count = self.read_u32()
+        patch_entries = {}
+        for _ in range(count):
+            hpath = self.read_fmt('<2L')[0]
+            btype = self.parse_bintype(self.read_u8())
+            string = self.read_string()
+            parts = string.split('.')
+            binvalue = self.read_field_basic(compute_binhash(parts[-1]), btype)
+            if hpath not in patch_entries:
+                if len(parts) > 1:
+                    patch_entries[hpath] = BinPtchEntry(hpath, [BinNestedField(compute_binhash(parts[0]), BinNested([]))])
+                else:
+                    patch_entries[hpath] = BinPtchEntry(hpath, [])
+
+            current_nesting = patch_entries[hpath]
+            for i in range(0, len(parts)-1):
+                if parts[i] not in current_nesting:
+                    current_nesting.fields.append(BinNestedField(compute_binhash(parts[i]), BinNested([])))
```

Comment:
You could define `__setitem__` on the base class (along with the existing `__getitem__`) for nicer way to set fields.